### ISSUE TITLE
Playback only once, then crash or error on next stream

### ISSUE THREAD

**Issue by chris42** [CastagnaIT/plugin.video.netflix#403]
#### Your Environment
- Netflix add-on version: 0.16.1
- Operating system version/name: LibreElec 9.2.0
- Device model: RPi4

Used Operating system:
* [ ] Android
* [ ] iOS
* [ ] Linux
* [ ] OSX
* [X] Raspberry-Pi
* [ ] Windows

### Describe the bug
When watching a series on my list, an episode will play fine, however only one/once. When selecting the next episode to play, I get a crash with the log below. A complete restart is needed, then I can watch another episode. Then, when electing the next one, I get the crash again.

#### Expected behavior
Next episode should play fine when selected

#### Actual behavior
Crash

#### Steps to reproduce the behavior

1. Select series from MyList
2. Select episode from series
3. Watch episode
4. Select next episode from series

### Debug log
The crash log is attached (removed profile ID)
[kodi_crashlog_20191229141840.log](https://github.com/CastagnaIT/plugin.video.netflix/files/4008079/kodi_crashlog_20191229141840.log)

### Additional Information
Widevine was updated today before crashes happened.

**Comment by CastagnaIT** [comment]
no need to remove IDs from the log, are only sha
the log missing all debug infos, read the readme how to

**Comment by chris42** [comment]
So, I turned on debug log. Of course it is huge, after running a full episode. However so far I only got an error when starting the next episode. Not a crash. Seems to point to some database stuff.
Let me know, if you need even more log.

[kodi.log](https://github.com/CastagnaIT/plugin.video.netflix/files/4008159/kodi.log)

**Comment by CastagnaIT** [comment]
you forgot to change debug to verbose, in the addon expert page setting

you don't necessarily have to watch a full-length movie you can skip to end
so take a new log

I have the impression that it is an error of a bugged sqlite library version used on libreelec
maybe I'll know more with a new log

**Comment by chris42** [comment]
Sorry, missed that. Attached the new log.
[kodi.log](https://github.com/CastagnaIT/plugin.video.netflix/files/4009612/kodi.log)

**Comment by CastagnaIT** [comment]
the first error is a "MemoryError", seems to be unable to process the data due to lack of ram

to now this happens only to you, this suggest that something doesn't work in your installation of libreelec or Libreelec have some problem of compatibility with RPI4

I don't know what to tell you

**Comment by jotakswe** [comment]
Hi!

I have about the same problem.
I'm using the x86-64 variant from LibreELEC in an HTPC.
If you want, I can post log file with debug enabled.
Unfortunately, I did not do so in my post on the LibreELEC forum, where I seek help for my problem.
https://forum.libreelec.tv/thread/21080-netflix-is-crashing/

**Comment by CastagnaIT** [comment]
> And I can add that the problem seems to come after I upgrade from 0.16.0 to 0.16.1.
if you say so you have to be 100% sure, comparing the two versions by installling from scratch, assume is not reliable...

I read the debug, to now i have no explanation for the cause, seems to strike libreelec version 9.2.0

**Comment by jotakswe** [comment]
I'm not 100% sure about this "after I upgrade from 0.16.0 to 0.16.1", but 99.9995% sure it applies to "libreelec version 9.2.0".

I'll fix a log for you, and we'll see what you find.

**Comment by chris42** [comment]
If it is a memory issue, could it be connected to the next two errors in my log, that are warning about a memory leak?

**Comment by jotakswe** [comment]
Here is my debug log.
[01_KODI.log](https://github.com/CastagnaIT/plugin.video.netflix/files/4013993/01_KODI.log)

**Comment by CastagnaIT** [comment]
[plugin.video.netflix_0.16.1_20200101_t1.zip](https://github.com/CastagnaIT/plugin.video.netflix/files/4014043/plugin.video.netflix_0.16.1_20200101_t1.zip)

can someone try installing this test?
I need to see if the type of error changes

Unfortunately in my virtual installation this error appeared only one time but now it doesn't happen anymore...

**Comment by jotakswe** [comment]
> 
> 
> [plugin.video.netflix_0.16.1_20200101_t1.zip](https://github.com/CastagnaIT/plugin.video.netflix/files/4014043/plugin.video.netflix_0.16.1_20200101_t1.zip)
> 
> can someone try installing this test?
> I need to see if the type of error changes
> 
> Unfortunately in my virtual installation this error appeared only one time but now it doesn't happen anymore...

I can test it.

**Comment by jotakswe** [comment]
First I want to say, Thank you for trying to fix this problem, it is much appreciated you should know.

Then, unfortunately, it did not work.
I managed to start 3 different movies in a row without problems, then I started 4 in a row that it did not work for, but became ERROR on.

[kodi_crashlog_20200101145643.log](https://github.com/CastagnaIT/plugin.video.netflix/files/4014150/kodi_crashlog_20200101145643.log)

**Comment by CastagnaIT** [comment]
I didn't say this was a fix

> I need to see if the type of error changes

I think i understand the problem but i'm not really sure
and i don't understand why it only happens on libreelec
I'll have to study a better solution before provide you a possible test for a solution

**Comment by CastagnaIT** [comment]
When you have time do a test with this and let me know what happen
[plugin.video.netflix_0.16.1_20200101_t2.zip](https://github.com/CastagnaIT/plugin.video.netflix/files/4014322/plugin.video.netflix_0.16.1_20200101_t2.zip)

**Comment by jotakswe** [comment]
And I haven't "accused" you of fixing the problem either, but I thanked you for trying. Is it difficult to receive some gratitude from someone who appreciates what you do?

Here is a new log file, but not after your new "fix".
And that's after I noticed that other programs have stopped working. Yesterday I tried to install an add-on to RSS, and the installation crashed and caused the computer to restart, and the program was not installed. Got to do the installation again, and then it worked. Tonight I started 2 movies in a row in Netflix, with your first test program, without any problems. But by the 3rd movie, things got wrong, and Netflix crashed, it popped up a log on the crash that I shut down, but the movie had started in the background despite the crash. After I saw the 4th movie, I turned off the movie at the end, and then it went wrong again, but this time it was the weather service Gismeteo that crashed and ended. Maybe you can find something in that log, but it's not a debug log.

Testing your last test program tomorrow.

[kodi_20200101222536.log](https://github.com/CastagnaIT/plugin.video.netflix/files/4014611/kodi_20200101222536.log)

**Comment by CastagnaIT** [comment]
your device have strange problems
but if it restarts on by itself, has some serious problems
it could be problem with mmc? (where you installed the os) or memory? there are suppositions i don't know, ...usually in computers unexpected reboots happen also for these reasons
the weather even if it doesn't work it is not a problem with nf

**Comment by jotakswe** [comment]
Hard to understand what you mean. Bad google translation ??
"the weather even if it doesn't work" ??
MMC, you mean flash memory? I have LibreELEC installed on a new SSD.
Do you want me to test the last test program you uploaded?

**Comment by CastagnaIT** [comment]
i am sorry but my english is not always good....
"the weather service Gismeteo even if it doesn't work it is not a problem with nf"
yes is most useful if you try the last test

**Comment by notoco** [comment]
We had a similar problem in one of the Polish forums. Raspberry just overheated. Changing the maximum resolution to 540 solved the problem.

**Comment by jotakswe** [comment]
My thoughts were if there could be problems with SQLite, which I think you mentioned earlier. Then there is no problem with NF, but it affects NF and other programs.
I will also run memory tests and hardware tests on my computer, which I have already done before installing Kodi for the first time. Maybe something happened after the installation, who knows.

**Comment by CastagnaIT** [comment]
before please do a test with the last test

**Comment by jotakswe** [comment]
Wow, that was a long time ago. :+1: 
I have now started 27-28 different films, left a review on one, thanks for changing to thumbs up and thumbs down, removed the same movie from "My list", started a Youtube video, just for testing, and not once, Netflix or something else has crashed. Wow, what did you do? :1st_place_medal: 

[kodi_crashlog_20200102101956.log](https://github.com/CastagnaIT/plugin.video.netflix/files/4015450/kodi_crashlog_20200102101956.log)

**Comment by jotakswe** [comment]
@chris42 Sorry for hijacking your bug report. Hopefully it can help you and your problems too.

**Comment by CastagnaIT** [comment]
it's called magic✨ haha

is a sqlite problem that is not a very good database system and sometimes it is a real nightmare
does not digest well multiple accesses to the database (probably in the most recent version of the module used in libreelec the problem becomes more difficult)

just to be safe, keep an eye out for the next few days and let me know if all it is ok

**Comment by jotakswe** [comment]
I will do that. :+1: 
Thank you so much for the help and the work you put into fixing this. :1st_place_medal:

**Comment by chris42** [comment]
I'll be back in the evening, will then run a test as well.

**Comment by jotakswe** [comment]
@CastagnaIT 
I have seen that there are other databases to install. Would that help to do that, and would Netflix use those databases instead? Is novice on Linux and Kodi, so you must have indulgence with it.

**Comment by CastagnaIT** [comment]
> I have seen that there are other databases to install. Would that help to do that, and would Netflix use those databases instead?

I don't think i understand what you're referring to...

**Comment by jotakswe** [comment]
MySQL or Maria DB, but it might not be the same type of databases?
As I said, I am novice in this context.

**Comment by CastagnaIT** [comment]
unfortunately they're not the same thing,
MySQL and forks like Maria DB, are elegant and high-performance solutions
but in order to work require a server and the connections fly with a tcp/ip protocol, so you can understand that it's not directly applicable

instead sqlite is serverless, so no server required, read and write directly from the hdd without other support software and no need to use a network infrastructure

**Comment by chris42** [comment]
Tried the last version and started a few streams. So far no error occurred.

**Comment by CastagnaIT** [comment]
can you confirm that the problem has not occurred again in the last few days?

**Comment by chris42** [comment]
Yep, so far no problem. Wanted to close the issue as soon as the new version is available.

**Comment by CastagnaIT** [comment]
Thanks, i think i will release a new version today or tomorrow

**Comment by alligatorshoes** [comment]
Just also wanted to provide my own feedback here. Using LibreElec 9.2.0 I was also getting *tons* of random crashes when trying to play a stream. Sometimes it would throw up the error box and sometimes Kodi would completely crash and restart.

Using your posted build above (plugin.video.netflix_0.16.1_20200101_t2.zip) I haven't had a single error or crash in days. Whatever you changed it seems to have fixed things - thanks so much!

**Comment by jotakswe** [comment]
@alligatorshoes  Same problem I had, error message or complete crashes.

@CastagnaIT  I still have some problems that need to be addressed.
Now it no longer works to rate or remove a movie from "My List".
It worked first, but then it stopped.

However, it has worked well to play movies.
No crashes at all.

[kodi_crashlog_20200105171504.log](https://github.com/CastagnaIT/plugin.video.netflix/files/4023932/kodi_crashlog_20200105171504.log)

**Comment by CastagnaIT** [comment]
tubmbs hadn't worked because of this error:
`HTTPError: 401 Client Error: Unauthorized for url: https://www.netflix.com/api/shakti/vb850f007/setThumbRating`
but this is another thing that's not about this issue,
to figure out how this error happens, i need to know the specific steps
if you know the steps to reproduce it you can open a new issue post

**Comment by doradojc** [comment]
> 
> 
> When you have time do a test with this and let me know what happen
> [plugin.video.netflix_0.16.1_20200101_t2.zip](https://github.com/CastagnaIT/plugin.video.netflix/files/4014322/plugin.video.netflix_0.16.1_20200101_t2.zip)

That worked for me, on a cleanly instalation of libreelec, thanks so much!

**Comment by CastagnaIT** [comment]
New version released
if necessary in the future, this issue can be reopened

**Comment by chris42** [comment]
Just tested 16.2, seems to be fixed.
